pkg/algorithm/nodeselect/v1alpha1/select_node.go

```diff
@@ -187,6 +210,96 @@ func (ac *Config) getBlockDevice() (*ndmapis.BlockDeviceList, error) {
 	if err != nil {
 		return nil, err
 	}
-	bdl := bdL.Filter(blockdevice.FilterClaimedDevices, blockdevice.FilterNonInactive)
+	bdl := bdL.Filter(blockdevice.FilterNonInactive)
 	return bdl.BlockDeviceList, nil
 }
+
+//TODO: Make changes in below code befor PR checked in
+// 1) Use Builder Pattern or some approach to present below code
+// 2) Refactor the below code before removing WIP
+
+// ClaimBlockDevice will create BDC for corresponding BD
+func (ac *Config) ClaimBlockDevice(nodeBDs *nodeBlockDevice, spc *apis.StoragePoolClaim) (*NodeClaimedBDDetails, error) {
```

Comment:
This function should just create BDCs for the BDs and done.
I would prefer to assign following responsibility to the function naively :
1. All the BDs  to be addressed are passed as args assuming only unclaimed BDs are passed.
2. Create claim for those
3. Done.

Reply:
Done as we discussed